### DISCUSSION TITLE
Q: What is the standard path for Eventum extensions?

### DISCUSSION THREAD

**Discussion by vladsf** [eventum/eventum#1205] [Q&A]
Hi,

What is the standard path for Eventum extensions? 
`eventum/config/extension` ?
`eventum/extension` ?
`some/path/outside/eventum` ?

Given the Upgrading manual, `eventum/config/extension` looks optimal, so the outside path as well.

**Comment by glensc** [comment]
There is no standard path, you can place it anywhere, just the line that points to extension class. must be accurate:
- https://github.com/eventum/eventum/blob/ff910e4f3195e3d06ddc39cca7d8821cf0433abc/docs/examples/extension/ExampleExtension.php#L28-L30

and if your extension has more classes, you define autoload for them:
- https://github.com/eventum/eventum/blob/ff910e4f3195e3d06ddc39cca7d8821cf0433abc/docs/examples/extension/ExampleExtension.php#L48-L53

my recommendation is to keep them outside eventum dir, so it's easier to perform upgrade of eventum.

**Reply by vladsf** [reply]
I can not get original ExampleExtension working anywhere outside `docs/examples/extension`. I get 
`PHP Fatal error:  Uncaught ReflectionException: Class Example\Auth\NullAuthAdapter does not exist in [..]/AuthController.php:108`

I fixed `$psr4` and `$baseDir` values but no luck.

**Reply by glensc** [reply]
You need to keep `NullAuthAdapter` relative to `ExampleExtension` or fix the autoload. Since I don't know what you did, can't comment more.

Meanwhile, you can read what psr-4 stands for:
- https://www.php-fig.org/psr/psr-4/

**Reply by vladsf** [reply]
Thank you . I fixed paths in `registerAutoloader()`
